python/TestHarness/TestHarness.py

```diff
@@ -1104,8 +1119,8 @@ def checkAndUpdateCLArgs(self):
         if opts.check_input and opts.enable_recover:
             print('ERROR: --check-input and --recover cannot be used simultaneously')
             sys.exit(1)
-        if opts.spec_file and not os.path.exists(opts.spec_file):
-            print('ERROR: --spec-file supplied but path does not exist')
+        if opts.run_dir and not os.path.exists(opts.run_dir):
+            print('ERROR: --run-dir does not accept a file name, please enter a directory instead')
```

Comment:
This error message is misleading: `exists` just checks that the path is valid, it doesn't differentiate between a directory and a file. So if you pass a directory that doesn't exist, you'll still get this error.